src/observableApiClient.ts

```diff
@@ -69,11 +70,18 @@ export class ObservableApiClient {
       throw error;
     }
 
-    if (response.status === 429) {
+    let rateLimitCount = 0;
+    while (response.status === 429) {
+      if (rateLimitCount >= MAX_RATE_LIMIT_ATTEMPTS) {
```

Comment:
It looks like `rateLimitCount` is only incremented once when `this._rateLimit === null`; am I reading it wrong? Normally I like to combine the increment and check into the same statement so that it’s more obvious that it can never infinite loop.

```suggestion
      if (rateLimitCount++ >= MAX_RATE_LIMIT_ATTEMPTS) {
```

Reply:
You're right that the rateLimitCount is only incremented when this._rateLimit is null. However, the bug is something different. `this._rateLimit` is intended to be used for multiple concurrent callers of this function to wait collaborative. That's not working with this new approach either though. I'll fix it up.